Lib/socket.py

```diff
@@ -127,7 +127,33 @@ def _intenum_converter(value, enum_klass):
     __all__.append("errorTab")
 
 
-class _GiveupOnSendfile(Exception): pass
```

Comment:
I honestly **don't like** this refactoring of `socket.py` just for the sake of reusing 10 lines of code in asyncio.  Someone could be using this `_check_sendfile_params` private API, and I don't see enough reasons to make any changes.

Please revert your changes here and just copy the code, it's fine.  It will also make it easier to study asyncio code, without the need to refer to `socket.py`.